test/unit/Trader.js

```diff
@@ -0,0 +1,99 @@
+const { expect } = require("chai")
+const { ethers, getNamedAccounts, deployments } = require("hardhat")
+const { deploy } = deployments
+const { signOrders } = require("@tracer-protocol/tracer-utils")
+
+describe("Unit tests: Trader.sol", function () {
+    let trader
+    let accounts
+    let perpMockAddress
+
+    before(async function () {
+        const { deployer } = await getNamedAccounts()
+
+        libPerpetuals = await deploy("Perpetuals", {
+            from: deployer,
+            log: true,
+        })
+
+        await deploy("Trader", {
+            from: deployer,
+            log: true,
+            libraries: {
+                Perpetuals: libPerpetuals.address,
+            },
+        })
+
+        await deploy("TracerPerpetualSwapMock", {
+            from: deployer,
+            log: true,
+        })
+
+        perpMockAddress = await deployments.get("TracerPerpetualSwapMock")
+
+        let traderDeployment = await deployments.get("Trader")
+        trader = await ethers.getContractAt(
+            traderDeployment.abi,
+            traderDeployment.address
+        )
+        accounts = await ethers.getSigners()
+    })
+
+    describe("executeTrader", async () => {
+        context("When the makers array is empty", async () => {
+            it("reverts", async () => {
+                await expect(trader.executeTrade([], [])).to.be.revertedWith(
+                    "TDR: Received empty arrays"
+                )
+            })
+        })
+        context("When the takers array is empty", async () => {
+            it("reverts", async () => {
+                await expect(trader.executeTrade([], [])).to.be.revertedWith(
+                    "TDR: Received empty arrays"
+                )
+            })
+        })
+        context("When the maker and taker array lengths differ", async () => {
+            it("reverts", async () => {
+                await expect(trader.executeTrade([], [])).to.be.revertedWith(
+                    "TDR: Received empty arrays"
+                )
+            })
+        })
```

Comment:
Just noting that while these might look like they are done, they will have to be re-written when you go to write the rest of the tests.